roles/network_plugin/flannel/templates/cni-flannel.yml.j2

```diff
@@ -36,11 +36,13 @@ data:
         "Type": "{{ flannel_backend_type }}"
       }
     }
+
+{% for arch in flannel_arches %}
 ---
 apiVersion: extensions/v1beta1
 kind: DaemonSet
 metadata:
-  name: kube-flannel
+  name: kube-flannel-ds-{{ arch }}
```

Comment:
When upgrading a cluster this can be a problem since we will end up with `kube-flannel` and `kube-flannel-ds-amd64` on the same nodes? Can we do something about this?

Reply:
Hmmm...not sure. Options:

(1) leave amd64 without -arch prefix and hope for the best
(2) make breaking change and put it into changelog 
(3) put multiarch flannel behind feature flag which is turned off by default.

Reply:
Would there be anything wrong with adding a task to delete the old daemonset (kube-flannel) after verifying the new one is running?  It seems that's the approach taken in a number of places in this codebase when upgrading components.

Reply:
@rhockenbury  can you give me an example where this approach is taken and I shall apply the same trick here?

Reply:
The coredns stuff might be a good example - https://github.com/kubernetes-sigs/kubespray/blob/master/roles/kubernetes-apps/ansible/tasks/cleanup_dns.yml

Reply:
Another option could be to add the arch prefix only not non amd64

Reply:
Ok, I think I added proper cleaning in pre-upgrade stage